Fix up 0.9.0 CHANGELOG (#2085)
spatialos/UnrealGDK: CHANGELOG.md

- We've introduced the feature flag `bEnableResultTypes` (default true). This configures interest queries to include only the set of components required for the queries to run. Depending on your game, this might save bandwidth.
- If you set the `bEnableResultTypes` flag to `true`, this disables dynamic interest overrides.
- We've moved the development authentication settings from the Runtime Settings panel to the Editor Settings panel.
- We've added the option to use the development authentication flow with the command line.
- We've added a button to generate a development authentication token inside the Unreal Editor. To use it, navigate to **Edit** > **Project Setting** > **SpatialOS GDK for Unreal** > **Editor Settings** > **Cloud Connection**.
- We've added a new section where you can configure the launch arguments for running a client on a mobile device. To use it, navigate to **Edit** > **Project Setting** > **SpatialOS GDK for Unreal** > **Editor Settings** > **Mobile**.

> - You can now choose which Runtime version to use when you launch a local or cloud deployment via the Spatial Runtime Settings.

Reviewer: ```suggestion
- You can now choose which Runtime version to use (in the Runtime Settings) when you launch a local or cloud deployment.
```

Updated to make it clear that you're choosing the Runtime version in the Runtime Settings, rather than launching a deployment via the Runtime Settings. Also removed "Spatial".

- The GDK now tracks properties that contain references to replicated Actors, so that it can resolve them again if the Actor that they reference moves out of and back into relevance.
- PIE sessions no longer occasionally fail to start due to missing schema for the `SpatialDebugger` Blueprint.
- We've fixed an issue where a newly-created subobject had empty state when `RepNotify` was called for a property pointing to that subobject.
- We've fixed an issue where deleted, initially dormant startup Actors would still be present on other worker instances.
- We now force-activate the RPC ring buffer when load balancing is enabled, to allow RPC handover when authority changes.
- We've fixed a race condition where a client that was leaving the deployment could leave its Actor behind on the server, to be cleaned up after a long timeout.
- We've fixed a crash that was caused by state in `SpatialGameInstance` persisting across a transition from one deployment to another.
- The GDK no longer crashes when you start and stop PIE clients multiple times.
- The GDK no longer crashes when shadow data is uninitialized when resolving unresolved objects.

> - We've fixed an occasional issue when sending component RPCs on a recently created actor.

Reviewer: ```suggestion
- We've fixed an issue that happened occasionally when sending component RPCs on a recently-created Actor.
```
